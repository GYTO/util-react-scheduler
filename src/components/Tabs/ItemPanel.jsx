// @flow
import * as React from 'react';
import { PRODUCT_ITEM} from '../../api/producs';
import type { ProductItem } from '../types/Products';

type Props = {
    selected: ?number
}

type State = {
    productItem: ProductItem[],
}

class ItemPanel extends React.Component<Props, State> {
    constructor(props: Props) {
        super(props);

        this.state = {
            productItem: PRODUCT_ITEM,
        };
    }

    render() {
        const { productItem } = this.state;
        const selectedItem = productItem.filter(item => {
            return item.id === this.props.selected;
        });

        return (
            <div>
                {this.props.selected ? selectedItem.map((item, index) => {
                    return <div key={index}>
                        <p>{item.id}</p>
                        <p>{item.name}</p>
                        <p>{item.text}</p>
                    </div>;
                }) : <p>Please select product tab</p>}
            </div>
        );
    }
}

export default ItemPanel;
