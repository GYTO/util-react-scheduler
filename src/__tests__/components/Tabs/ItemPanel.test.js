import * as React from 'react';
import {shallow, mount} from 'enzyme';
import ItemPanel from '../../../components/Tabs/ItemPanel';

describe('ItemPanel', () => {
    let props;
    let shallowRender;
    let mountedRender;

    const getMounted = () => {
        if (!mountedRender) mountedRender = mount(<ItemPanel {...props} />);
        return mountedRender;
    };

    const getShallow = () => {
        if (!shallowRender) shallowRender = shallow(<ItemPanel {...props} />);
        return shallowRender;
    };

    beforeEach(() => {
        jest.clearAllMocks();

        props = {
            selected: null,
        };

        shallowRender = undefined;
        mountedRender = undefined;
    });

    it('matches snapshot', () => {
        expect(getShallow()).toMatchSnapshot();
    });

    it('renders component and expect regular text', () => {
        expect(getMounted().find('p').first().text()).toEqual('Please select product tab')
    });

    it('not renders regular text ', () => {
        props.selected = 1;
        console.log(getMounted().debug());
        expect(getMounted().find('p').first().text()).not.toEqual('Please select product tab');
        expect(getMounted().prop('selected')).toEqual(1);
    });
});
