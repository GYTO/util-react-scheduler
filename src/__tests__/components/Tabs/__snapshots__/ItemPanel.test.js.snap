// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemPanel matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItemPanel
    selected={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p>
        Please select product tab
      </p>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Please select product tab",
      },
      "ref": null,
      "rendered": "Please select product tab",
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          Please select product tab
        </p>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Please select product tab",
        },
        "ref": null,
        "rendered": "Please select product tab",
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
